internal/kuberesource/sets.go

```diff
@@ -698,3 +698,138 @@ func GPU() []any {
 
 	return []any{tester}
 }
+
+// Vault returns the resources for deploying a user managed vault.
+func Vault(namespace string) []any {
+	// The bao operator init command should not be scripted like this in a real production environment,
+	// because it will leak the recovery keys to anyone with kubernetes log access.
+	// Normally this should be executed by an admin setting up the Vault once and the
+	// recovery keys should be stored securely.
+	baoInitCmd := `echo "Waiting for Vault to become available..."
+# Wait until Vault server instance is ready
+until wget --server-response --no-check-certificate --spider -q -O- https://vault:8200 2>&1 | grep -q "HTTP/1.1 200"; do
+    sleep 2
+done
+
+
+# Get init status using wget
+echo "Vault is responding. Checking initialization status..."
+init_check=$(wget --no-check-certificate -q -O - https://vault:8200/v1/sys/init)
+
+# Check if Vault is uninitialized
+if echo "$init_check" | grep -q '"initialized"[ ]*:[ ]*false'; then
```

Comment:
This might actually be easier to understand if inlined:

```sh
if wget --no-check-certificate -q -O - https://vault:8200/v1/sys/init | grep -q '"initialized"[ ]*:[ ]*false'; then
```